### PR TITLE
[BEAM-1921] expose connectionProperties in JdbcIO

### PR DESCRIPTION
**Pull request by XuMingmin**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @jbonofre

File: sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java [hunk 1]
```diff
@@ -217,6 +219,20 @@ public DataSourceConfiguration withPassword(String password) {
...
+     * Sets the connection properties passed to driver.connect(...).
+     * Format of the string must be [propertyName=property;]*
+     *
+     * <p>NOTE - The "user" and "password" properties can be add via {@link #withUsername(String)},
+     * {@link #withPassword(String)}, so they do not need to be included here.
+     * @param connectionProperties
+     * @return
+     */
+    public DataSourceConfiguration withConnectionProperties(String connectionProperties) {
+      checkArgument(connectionProperties != null, "connectionProperties is null");
```
Comment (jbonofre) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java]: Can you use the same kind of message as for the other `with*` methods ? Something like `DataSourceConfiguration.create(driver, url).withConnectionProperties(connectionProperties) called with null connectionProperties)`.

Comment (XuMingmin) [inline sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java]: +1

**Comment by XuMingmin** [comment]
Thanks @jbonofre for the review!

**Pull request closed without merging**